src/lib.rs

```diff
@@ -1072,13 +1073,18 @@ impl CecConnection {
         }
     }
 
-    pub fn get_device_power_status(&self, address: CecLogicalAddress) -> CecConnectionResult<()> {
-        if unsafe { libcec_get_device_power_status(self.1, address.into()) } == 0 {
-            Err(CecConnectionResultError::TransmitFailed)
-        } else {
-            Ok(())
+    pub fn get_device_power_status(&self, address: CecLogicalAddress) -> CecPowerStatus {
+        let status_raw: cec_power_status =
+            unsafe { libcec_get_device_power_status(self.1, address.into()) };
+        match CecPowerStatus::try_from(status_raw) {
+            Ok(status) => status,
+            Err(status_raw) => {
+                warn!("get_device_power_status: Could not convert result {} to rust enum. Returning Unknown", status_raw);
+                CecPowerStatus::Unknown
+            }
         }
```

Comment:
I copied the pattern used in `get_active_source` (above) here.

Reply:
Sounds good, Unknown should be good enough fallback.